reflection.md

```diff
@@ -2,17 +2,51 @@
+
 ## Comment on the design of this internal DSL
 
 Write a few brief paragraphs that discuss:
    + What works about this design? (For example, what things seem easy and
    natural to say, using the DSL?)
+   
+   It's much more convenient to chain literals together in a union. Under the previous language, the user had to specify the exact "associative order" of what pairs of literals are put into a union first, before which other pairs of literals are added on as unions. It's much simpler just to say '0' || '1' || '2' || '3' || '4' || '5' || '6' || '7' || '8' || '9' and not have to specify the order that characters are chained together in an expression that expresses a union of elements. 
+   
+   I also think the implementation of (some RegularExpression){some integer} to represent repetition is clever and intuitive.
+   
+   I think that <*> (despite it's unavoidably cumbersome < >, which I will discuss later) is a great method name to represent a regular expression that matches zero or more repetitions of another. '*' is usually the "wild card" symbol, and it's used in Google search, for example, when you have a specific query in mind but would also want to perform searches that are a continuation of your specific query. For example, if you wanted to search "engineer" but also wanted the search results to return information about "engineers" and "engineering", you could then edit your query to be "engineer*" . Even though the meaning of * doesn't exactly translate to the meaning of <*> in our language, I personally do see an intutive connection.
+   
+   
+ 
    + What doesn't work about this design? (For example, what things seem
    cumbersome to say?)
+   
+What seems a bit more cumbersome to say the following: " val telNumber = '(' ~ digit{3} ~ ')' ~ digit{3} ~ '-' ~ digit{4} ". However, given the flexibility and constraints we want users to be able to follow in creating a valid phone number, I honestly cannot think of a simpler solution that does not compromise either the flexibility and/or the constraints. The flexibility includes being able to choose any 3 digits between the parentheses, to choose any of the 3 digits before the dash, and to choose any 4 digits after the dash. The constraints include making sure that only (, ), -, and digits are included. Given the specific function of each element in the series of characters that compose a valid tedNumber, I can't think of any simpler way to say this (although this isn't a critique of the design -- this is my way to express how the design is "cumbersome" but as simple as I think it can get). Overall, the < > surrounding * and surrounding + make the language look cumbersome, but then again, I really can't come up with an alternative way to express what <*> and <+> express, and we definitely can't just remove the < > because just having * totally changes the meaning of <*>
```

Comment:
Agreed&mdash;it would be an interesting idea to be able to define a template string like 
`(ddd)-ddd-dddd` that gets turned into what you have above (or event `(123)-456-7890`). That's one approach we could take to simplify writing it, but it does, as you point out, come at the cost of flexibility.